### PR TITLE
Add SkipClean option

### PR DESCRIPTION
**Pull request by jingweno**
This PR rebases https://github.com/gorilla/mux/pull/54 on the latest master. Most works were done by @whatupdave.

/cc @elithrar

### CODE REVIEW

File: route.go [hunk 1]
```diff
@@ -26,6 +26,9 @@ type Route struct {
 	// If true, when the path pattern is "/path/", accessing "/path" will
 	// redirect to the former and vice versa.
 	strictSlash bool
+	// If true, when the path pattern is "/path//to", accessing "/path//to"
+	// will not redirect
+	skipClean bool
```
Comment (elithrar) [inline route.go]: We should still add `SkipClean()` as a method on `*Route` as well.

File: mux.go [hunk 2]
```diff
@@ -133,6 +137,19 @@ func (r *Router) StrictSlash(value bool) *Router {
 	return r
 }
 
+// SkipClean defines the path cleaning behaviour for new routes. The initial
+// value is false.
```
Comment (elithrar) [inline mux.go]: `SkipClean defines the path cleaning behaviour for new routes. The initial value is false. Users should be careful about which routes are not cleaned`.

**Comment by elithrar** [comment]
Ping @jingweno in case you didn't see my line notes.

**Comment by jingweno** [comment]
@elithrar Sorry for the late response. I addressed all comments and added a test. Could we merge this if it looks okay? I'd need the change to fix a production issue. Thanks.

**Comment by jingweno** [comment]
This closes https://github.com/gorilla/mux/pull/54.

**Comment by jingweno** [comment]
I removed installing `go-vet` as part of the build since it's now part of Go std distro: https://travis-ci.org/gorilla/mux/builds/127283267.

**Comment by elithrar** [comment]
Sure, looking at it now!

**Comment by jingweno** [comment]
Thanks @elithrar!

**Comment by elithrar** [comment]
LGTM. Thansk @jingweno!

**Pull request merged**